### DISCUSSION TITLE
On opening the combobox, if there are 100 or more items, the selection defaults to the first item

### DISCUSSION THREAD

**Discussion by ivantoniev** [josefarias/hotwire_combobox#230] [Q&A] [answered]
Hello, thanks for the great gem!

We have a few hundred items in our combobox and we found it strange that when opened if you're not doing a new selection and [there' more than 100 items](https://github.com/josefarias/hotwire_combobox/blob/main/app/assets/javascripts/hotwire_combobox.esm.js#L1150), the combobox will default to the first item of the list.
What is the reason for this behaviour? I couldn't dig it out from the commit history. Our expectation would have been the original selection will remain.

**Comment by LouisKottmann** [comment]
I've just been bitten by this very behavior, which was counterintuitive to track down.

The performance decision should be up to me I think, maybe let us customize the number?

As is stands, this single hardcoded limit implies that pagination is not an option for dynamic collections, it is a hard requirement.

I am yet to give an honest try with pagination and async, and will report on that once I do in the coming days.

Thanks for the work !

**Comment by LouisKottmann** [comment]
I went on and tried to use the async version, which works when elements sent are limited to less than 100.

I did not try pagination but I figure it should work as well up to 100 elements, which remains a problem.

For reference, I mean this code: https://github.com/josefarias/hotwire_combobox/blob/db65819a40078c4c0e17fc0cba1007f2426afe83/app/assets/javascripts/hw_combobox/models/combobox/selection.js#L88

However, I found no obvious way to make hotwire_combobox work async on a non-AR collection with regards to the preselected value (I pass in the ID, but there is no way to pass in the display value). 
This (closed) issue is related: https://github.com/josefarias/hotwire_combobox/issues/123 in particular on the "it only works if display value and value are the same".

I will make a repo that showcases this.

EDIT: https://github.com/josefarias/hotwire_combobox/issues/247

**Comment by josefarias** [comment, marked as answer]
Hi @ivantoniev, @LouisKottmann

Limit removed in https://github.com/josefarias/hotwire_combobox/pull/248, original reasoning explained in the description.

If you don’t mind me asking, what kinds of collections are you rendering? Almost all the use cases I’ve encountered so far benefit from only rendering a subset of options at once. Curious if that’s not the case for you, or maybe it’s too cumbersome to render a subset?

In any case, you shouldn’t have that problem once the PR is merged and you point your library to the latest commit.

**Comment by LouisKottmann** [comment]
I'm rendering a collection of customer companies (id => name) to select users.

The view is not meant to receive wide traffic and thus not prioritized for optimization.

I am also dealing with a client that strongly dislikes pagination, and so `pagy` and the like are not part of the stack (yet).

I will add pagy and try it out to verify that once I fetch enough pages to reach the treshold of 100, the combobox is still showing the selected element.

**Comment by LouisKottmann** [comment]
@josefarias my suspicions panned out, when using a paginated async combobox, and selecting an element past the first 100 elements (so, scrolling and getting enough pages to have over 100 total elements, then selecting one in the bottom), the previously selected element is no longer highlighted when we get to it, and closing the combobox without making a new selection actually falls back to selecting the first element.

Should I write a new issue for this?

**Reply by josefarias** [reply]
@LouisKottmann I’m afraid I don’t quite understand. Yes, a new issue and reproduction example would be great, please.

**Reply by LouisKottmann** [reply]
Actually, it will be fixed by #248 so nevermind, sorry, I should have taken that change into account.
